src/venstarcolortouch/venstarcolortouch.py

```diff
@@ -211,9 +218,14 @@ def update_info(self):
         # T2xxx, T3xxx thermostats (and maybe more) always use Celsius in the API regardless of the display units
         # So handle this case accordingly
         if self.model.startswith(("T2", "T3")):
-            # Always degC
-            self.tempunits = self.TEMPUNITS_C
-            logging.debug("Detected thermostat model %s, using temp units of Celsius", self.model)
+            # Always degC for firmware <= 5.28
+            if self.get_firmware_ver() < UNIT_BUG_FIX_VERSION:
+                self.tempunits = self.TEMPUNITS_C
+                logging.debug("Detected thermostat model %s, using temp units of Celsius", self.model)
+            else:
+                self.tempunits = self.display_tempunits
+                logging.debug("Detected thermostat model %s, but new firmware(%s) fixes units bug",
+                              self.model, self.get_firmware_ver())
```

Comment:
Instead of nesting conditionals and adding all this code here, you could add the version check to line 220 and include the T2/T3's on 239

```
if self.model.startswith(("T2", "T3")) and self.get_firmware_ver() < UNIT_BUG_FIX_VERSION:
    # Always degC due to bug in older FW versions 
    self.tempunits = self.TEMPUNITS_C
    logging.debug("Detected thermostat model %s, using temp units of Celsius", self.model)
elif self.model in ["VYG-4900-VEN", "VYG-4800-VEN", "VYG-3900", "COLORTOUCH"] or self.model.startswith(("T2", "T3")):
    # Same as display units
    self.tempunits = self.display_tempunits
```

Reply:
I pushed these changes.